### ISSUE TITLE
[feature][nested functions support] Can be implemented by mapping Python's nested functions to C++

### ISSUE THREAD

**Issue by yurivict** [shedskin/shedskin#474]
C++ can have nested functions which can be implemented by using local lambda functions.
Python's nested functions can be nicely mapped to C++ nested lambda functions.

**Comment by srepmub** [comment]
yeah, but it probably also requires quite a bit of work on the inference side.. which is already quite complicated.

support for closures is in a similar situation.

I don't expect to work myself on most of the currently documented limitations, at least in the near future.
